### ISSUE TITLE
Use cursor as "source of truth" for where a dropped item lands

### ISSUE THREAD

**Issue by tcc-sejohnson** [isaacHagoel/svelte-dnd-action#397]
Hey Isaac,

While playing around with this library, one frustration I've had is with where items land when dragged. Ideally, I'd like a happy medium between `centreDraggedOnCursor: true` and `centreDraggedOnCursor: false`. What do I mean by this?

### `centreDraggedOnCursor: true`

With this set, dragging an element centers the element on the cursor. This has the benefit of making sure that _wherever your cursor is when the drag finishes is the container the element is dropped into_. However, this can be really weird. If I grab a wide element near the left of the screen, the element will center itself on the cursor, putting half of itself offscreen. This isn't super-desirable behavior, as it can suddenly "teleport" a dragged item from its original position to centered on your cursor's position.

### `centreDraggedOnCursor: false`

With this, elements don't haphazardly jump to your cursor when you start dragging them, but which DnD zone they fall into can be a bit confusing. For example, if I grab the left side of an item and try to drag it into a DnD zone to the left, it's possible it will fall back into the right zone because the dragged item isn't far enough over the left DnD zone to fall into it (even though my cursor is over the left DnD zone).

### The Solution

I think the ideal solution is a new option called something like `selectZoneBasedOnCursor` or `dropOnCursor`. This would tell `svelte-dnd-action` that I want to drop my dragged item into whichever DnD zone my _cursor_ is over, but it doesn't require the dragged item to center itself on my cursor.

I'm going to leave this here for you to discuss -- if it's something we'd like, I'm happy to look into opening a PR for it.

**Comment by isaacHagoel** [comment]
sorry, i was sure i replied to it months ago. I am open to a PR that makes this an optional behaviour, but don't expect it to be super easy to do. 
the current implementation uses the center of the element because in the majority of cases it leads to a more pleasant behaviour.

**Comment by jessmartin** [comment]
I agree with @tcc-sejohnson! This is my desired behavior. Check out the behavior of dragging a card in Trello. It works the way that @tcc-sejohnson describes. 

If you implement it, I'll use it @tcc-sejohnson. 😉

**Comment by yadoga** [comment]
would be nice, @tcc-sejohnson !

**Comment by tcc-sejohnson** [comment]
Wish we had a !remindme bot like Reddit 😉 I'm on vacation currently, but would love to make time to work on this.

**Comment by jessmartin** [comment]
@tcc-sejohnson This is just me pretending to be your !remindme bot. 😜 I'm still interested in this functionality-would be happy to test it if you got a version of it working!

**Comment by its-all-waves** [comment]
I would love this feature! Definitely is my desired behavior. The width of a draggable item can vary greatly, so it's best to drop it where the pointer lands, whether that's the cursor or a finger. It especially makes sense in the context of a finger.

**Comment by isaacHagoel** [comment]
Closing as superseded by fixes and APIs now present in v0.9.77. If this still reproduces on v0.9.77, please open a focused issue with current versions and a minimal reproduction.
